src/tensor_operations/matrix_decomposition.jl

```diff
@@ -531,10 +535,28 @@ function factorize_qr(A::ITensor, Linds...; kwargs...)
   return L, R
 end
 
-function factorize_svd(A::ITensor, Linds...; kwargs...)
+#
+# Generic function implementing a square root decomposition of a diagonal, order 2 tensor with inds u, v
+#
+function sqrt_decomp(D::ITensor, u::Index, v::Index)
+  (storage(D) isa Union{Diag,DiagBlockSparse}) ||
+    error("Must be a diagonal matrix ITensor.")
+  sqrtDL = diagITensor(u, dag(u)')
+  sqrtDR = diagITensor(v, dag(v)')
+  map_diag!(sqrt ∘ abs, sqrtDL, D)
+  map_diag!(sqrt ∘ abs, sqrtDR, D)
+  δᵤᵥ = copy(D)
+  map_diag!(sign, δᵤᵥ, D)
+  return sqrtDL, prime(δᵤᵥ), sqrtDR
+end
+
+function factorize_svd(A::ITensor, Linds...; (singular_values!)=nothing, kwargs...)
   ortho::String = get(kwargs, :ortho, "left")
   alg::String = get(kwargs, :svd_alg, "divide_and_conquer")
-  USV = svd(A, Linds...; kwargs..., alg=alg)
+  dir = get(kwargs, :dir, ITensors.In)
```

Comment:
Let's move this as well as `ortho` and `alg` into the function signature since we are trying to move towards that style in general (as opposed to using `get`).